### PR TITLE
[RC-1] Required inner message validation

### PR DESCRIPTION
**Pull request by rogchap**
Adds validation rule for required inner message.

```proto
message Profile {
  string id = 1;
  Address address = 2 [(validator.msg_required) = true];
}
```

### CODE REVIEW

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 1]
```diff
@@ -409,6 +420,10 @@ func getLengthGteValue(field *descriptor.FieldDescriptorProto) *int64 {
 	return nil
 }
 
+func getMsgRequiredValue(field *descriptor.FieldDescriptorProto) bool {
```
Comment (chenzhihao) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go]:
```suggestion
func IsMsgRequired(field *descriptor.FieldDescriptorProto) bool {
```

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 2]
```diff
@@ -242,6 +242,16 @@ func (p *plugin) generateLengthValidator(variableName string, ccTypeName string,
...
+		if nullable {
+			p.P(`if `, variableName, ` == nil {`)
+			p.In()
+			p.P(`return fmt.Errorf("message `, fieldName, ` is required")`)
+			p.Out()
+			p.P(`}`)
+		}
+	}
+
 	if nullable {
```
Comment (chenzhihao) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go]: How about moving the above logic into this block

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 3]
```diff
@@ -242,6 +242,16 @@ func (p *plugin) generateLengthValidator(variableName string, ccTypeName string,
...
 		p.P(`if `, variableName, ` != nil {`)
 		p.In()
```
Comment (chenzhihao) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go]:
```suggestion
		p.In()
		if getMsgRequiredValue(field) {
		    p.P(`return fmt.Errorf("message `, fieldName, ` is required")`)
		    p.Out()
		    p.P(`}`)
		}	
```

Comment (rogchap) [reply protobuf/protoc-gen-govalidator/plugin/plugin.go]: This is already past the `!= nil` check. This won't output the correct code.

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 4]
```diff
@@ -243,6 +243,14 @@ func (p *plugin) generateLengthValidator(variableName string, ccTypeName string,
 func (p *plugin) generateInnerMessageValidator(variableName string, ccTypeName string, fieldName string, field *descriptor.FieldDescriptorProto, nullable bool) {
 
 	if nullable {
+		if isMsgRequired(field) {
+			p.P(`if `, variableName, ` == nil {`)
+			p.In()
+			p.P(`return fmt.Errorf("message `, fieldName, ` is required")`)
+			p.Out()
+			p.P(`}`)
+		}
```
Comment (chenzhihao) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go]: `else` ?

Comment (rogchap) [reply protobuf/protoc-gen-govalidator/plugin/plugin.go]: I need both code blocks to be outputted.

**Pull request merged**